modules/configuration/secret.tf

```
@@ -0,0 +1,32 @@
+resource "duplocloud_k8_secret" "managed" {
+  count              = var.managed && var.class == "secret" ? 1 : 0
+  tenant_id          = var.tenant_id
+  secret_name        = local.name
+  secret_type        = "Opaque"
+  secret_data        = local.data
+  secret_annotations = local.annotations
+  timeouts {
+    create = "3m"
+    update = "3m"
+    delete = "3m"
+  }
+}
+
+resource "duplocloud_k8_secret" "unmanaged" {
+  count              = var.managed && var.class == "secret" ? 1 : 0
```

```suggestion
  count              = var.managed && var.class == "secret" ? 0 : 1
```
currently the count condition is the same for managed and unmanaged, so it would be creating both?

---

ohh I see you did it like this further down for configmap
`count     = !var.managed && var.class == "configmap" ? 1 : 0`
I'm fine with either style

---

Oh you caught a missing not operator there. Fixed that up.